Feat(web-react): Introduce `ActionGroup` component #DS-1621 (#1843)
lmc-eu/spirit-design-system: packages/web-react/src/components/ActionGroup/README.md

# ActionGroup

This component uses the `Flex` component internally.

> For more customization options, refer to the props available in the [Flex documentation][web-react-flex-documentation].

Reviewer: ```suggestion
For more customisation options, refer to the props in the [Flex documentation][web-react-flex-documentation].
```

ActionGroup component is a container component that is used to position action components in a predefined layout.

> - For desktop and tablet views, actions are positioned in a horizontal direction with reverse order.

Reviewer: ```suggestion
- For desktop and tablet views, actions are positioned in a horizontal direction in reverse order.
```

- For mobile view, actions are positioned in a vertical layout and are stretched to the full width of the container.

```jsx
<ActionGroup>

>   <!-- Actions components goes here -->

Reviewer: ```suggestion
  <!-- Actions components go here -->
```

</ActionGroup>
```

## Default Props

ActionGroup comes with predefined default props designed to simplify its use.

> These defaults can be customized as needed to suit your requirements.

Reviewer: ```suggestion
These defaults can be customised as needed to suit your requirements.
```

| Name       | Type                                                                | Value                                                    |
| ---------- | ------------------------------------------------------------------- | -------------------------------------------------------- |

> | alignmentX | \[[AlignmentX dictionary][alignment-dictionary] \| `object`]        | `{{ mobile: 'stretch', tablet: 'left' }}`                |
> | direction  | \[[DirectionExtended dictionary][direction-dictionary] \| `object`] | `{{ mobile: 'vertical', tablet: 'horizontal-reverse' }}` |

Reviewer 2: Our PM expects the `alignmentX` to be the same on mobile and desktop when all you set is the direction to `horizontal-reversed`:

```jsx
<ActionGroup direction="horizontal-reversed">
```

However, with current defaults, you would end up with actions on the right on mobile (because of `stretch`) and on the left on tablet and desktop (because of `left`).

So, having it discussed in person, I think we need to adjust the default `alignmentX` like this:

```suggestion
| alignmentX | \[[AlignmentX dictionary][alignment-dictionary] \| `object`]        | `stretch`                |
| direction  | \[[DirectionExtended dictionary][direction-dictionary] \| `object`] | `{{ mobile: 'vertical', tablet: 'horizontal-reverse' }}` |
```

Author: > So, having it discussed in person, I think we need to set the default `alignmentX` to `stretch` for all breakpoints

But then it will always be aligned to the right on desktop and tablet
